### PR TITLE
[OpenTelemetry.Instrumentation.AWSLambda]: consider two sources when setting http.method and http.target: original request and request context.

### PR DESCRIPTION
**Pull request by rypdal**
**Problem**

For AWS Lambda functions (V1 only) invoked via HTTP depending on setup there might be different sources for setting http.method and http.target: original request (APIGatewayProxyRequest) and request context (ProxyRequestContext) defined inside of the request.

**Fix**

Consider Path and HttpMethod properties of APIGatewayProxyRequest and APIGatewayProxyRequest.ProxyRequestContext when setting http.target and http.method. Request context should be prioritised over request while setting http.target and http.method

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaHttpUtils.cs [hunk 1]
```diff
@@ -44,8 +44,9 @@ internal class AWSLambdaHttpUtils
         {
             case APIGatewayProxyRequest request:
                 httpScheme = AWSLambdaUtils.GetHeaderValues(request, HeaderXForwardedProto)?.LastOrDefault();
-                httpTarget = string.Concat(request.RequestContext?.Path ?? string.Empty, GetQueryString(request));
-                httpMethod = request.HttpMethod;
+                var path = request.Path ?? request.RequestContext?.Path ?? string.Empty;
+                httpTarget = string.Concat(path, GetQueryString(request));
+                httpMethod = request.HttpMethod ?? request.RequestContext?.HttpMethod;
```
Comment (Oberon00) [inline src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaHttpUtils.cs:49]: Should we also consider the case where the method is null/empty? I guess in that case it does not make sense to set any (other) http attributes

**Review by Oberon00 — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaHttpUtils.cs (hunk 1 above)
Comment (rypdal) [reply src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaHttpUtils.cs:49]: I would consider it more general: should we set all required tags of one (or more) of required are missing ? But then it makes code more complex. I think we still should be more tolerant and set as more as we can.

**Review by Oberon00 — CHANGES_REQUESTED** [review]

File: src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaHttpUtils.cs [hunk 2]
```diff
@@ -44,8 +44,9 @@ internal class AWSLambdaHttpUtils
         {
             case APIGatewayProxyRequest request:
                 httpScheme = AWSLambdaUtils.GetHeaderValues(request, HeaderXForwardedProto)?.LastOrDefault();
-                httpTarget = string.Concat(request.RequestContext?.Path ?? string.Empty, GetQueryString(request));
-                httpMethod = request.HttpMethod;
+                var path = request.Path ?? request.RequestContext?.Path ?? string.Empty;
```
[the remark this answers is not in the document]
Comment (rypdal) [reply src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaHttpUtils.cs]: I swapped setting and mention prioritisation in the PR description.

**Review by Oberon00 — APPROVED** [review]

**Review by utpilla — APPROVED** [review]
LGTM overall. Resorting to the component owner's approval for the specific changes made in the PR.

**Pull request merged**